src/functional/iterator/embedded.py

```diff
@@ -100,16 +96,21 @@ def __getitem__(self, indices: tuple[int, int]) -> int:
 # Offsets
 OffsetPart: TypeAlias = Tag | IntIndex
 CompleteOffset: TypeAlias = tuple[Tag, IntIndex]
-OffsetProviderElem: TypeAlias = Dimension | Connectivity
+OffsetProviderElem: TypeAlias = (
+    common.Dimension
+    | common.Connectivity
+    | NeighborTableOffsetProvider
+    | StridedNeighborOffsetProvider
```

Comment:
Is this a mypy issue? We should not have to mention every implementation of a Connectivity here.

Reply:
It is an issue of attributes being accessed on things that are typed as `Connectivity` which are only defined in these subclasses. If you think these attributes should be defined at the base class level then we should go for that. Other solutions might be to introduce another level in the hierarchy in between, or the use of a protocol.

I didn't have the time to get into evaluating design changes.